lib/ecto/json.ex

```diff
@@ -46,3 +46,52 @@ if Code.ensure_loaded?(Jason.Encoder) do
     end
   end
 end
+
+if Code.ensure_loaded?(JSON.Encoder) do
+  defimpl JSON.Encoder, for: Ecto.Association.NotLoaded do
+    def encode(%{__owner__: owner, __field__: field}, _) do
+      raise """
+      cannot encode association #{inspect(field)} from #{inspect(owner)} to \
+      JSON because the association was not loaded.
+
+      You can either preload the association:
```

Comment:
Maybe could avoid the code duplication by making both JSON/Jason implementations share code? 

e.g. with `for protocol  <- [JSON.Encoder, Jason.Encoder]`